### ISSUE TITLE
[BUG] SOAP service client's debugContainer leaks file descriptors

### ISSUE THREAD

**Issue by brakthehack** [vmware/govmomi#3154]
**Describe the bug**
When  Service clients are created when `vim25`'s `debug` file provider is set, file descriptors may leak if the service client goes out of scope.

**To Reproduce**

### Set the FileProvider enabling govmomi tracing.
```
import "github.com/vmware/govmomi/vim25/debug"
        tmpdir, _ := os.MkdirTemp("", "")
	fp := debug.FileProvider{
		Path: tmpdir,
	}

	debug.SetProvider(&fp)
```

### Instantiate a service client. For example:
```
// NewClient returns a new EAM client.
func NewClient(c *vim25.Client) *Client {
	return &Client{
		Client: c.Client.NewServiceClient(Path, Namespace),
	}
}
```

### Make some requests.
```
client := eam.NewClient(govmomiClient.Client)
req := eamtypes.QueryAgent{
    This: *agency,
}
res, err := methods.QueryAgent(ctx, client, &req)
if err != nil {
    return nil, err
}
```

At this point the client can go out of scope and it is my understanding that this is expected as service clients are meant to be somewhat ephemeral as they multiplex their requests with other service clients over the underlying govmomi (and http) transport. One should notice there is no `Close` method exposed for service clients.

Notice that after the client goes out of scope an `lsof` output shows the process still holding onto the FD for the client:
```
test 18658  root  329u      REG               0,40       377 16619337 /tmp/1-client.log
```

The file is [opened](https://github.com/vmware/govmomi/blob/aa3f6d6a6456547703ed40b0735cc0aaaf473724/vim25/soap/debug.go#L138) when `NewClient` is called and the file handle is never released. Each request is written to the file. Because the file is never closed, we leak FDs.

**Expected behavior**
After a request is made, the service client should [close](https://github.com/vmware/govmomi/blob/aa3f6d6a6456547703ed40b0735cc0aaaf473724/vim25/soap/client.go#L567), the FD and reopen it as needed for new requests.

**Affected version**
ToT

**Screenshots/Debug Output**
Snip of `lsof` output:
```
1792264305458/27-client.log
testsvc  18658  test   79u      REG               0,40        81 16500618 /tmp/test-18658-2023-06-07T05-56-09.2573311792264305458/174-client.log
testsvc  18658  test   80u      REG               0,40       153 16487765 /tmp/test-18658-2023-06-07T05-56-09.2573311792264305458/28-client.log
testsvc  18658  test   81u      REG               0,40       440 19933180 /tmp/test-18658-2023-06-07T05-56-09.2573311792264305458/4-11444.req.xml
testsvc  18658  test   82u      REG               0,40       518 16484830 /tmp/test-18658-2023-06-07T05-56-09.2573311792264305458/29-client.log
testsvc  18658  test   83u      REG               0,40        81 16514772 /tmp/test-18658-2023-06-07T05-56-09.2573311792264305458/192-client.log
testsvc  18658  test   84u      REG               0,40       153 16476877 /tmp/test-18658-2023-06-07T05-56-09.2573311792264305458/30-client.log
testsvc  18658  test   85u      REG               0,40       977 16499546 /tmp/test-18658-2023-06-07T05-56-09.2573311792264305458/175-client.log
testsvc  18658  test   86u      REG               0,40       169 16462793 /tmp/test-18658-2023-06-07T05-56-09.2573311792264305458/36-client.log
testsvc  18658  test   87u      REG               0,40       153 16462776 /tmp/test-18658-2023-06-07T05-56-09.2573311792264305458/31-client.log
testsvc  18658  test   88u      REG               0,40       977 16521811 /tmp/test-18658-2023-06-07T05-56-09.2573311792264305458/193-client.log
testsvc  18658  test   89u      REG               0,40       153 16480078 /tmp/test-18658-2023-06-07T05-56-09.2573311792264305458/32-client.log
testsvc  18658  test   90u      REG               0,40       977 16510652 /tmp/test-18658-2023-06-07T05-56-09.2573311792264305458/176-client.log
testsvc  18658  test   91u      REG               0,40        88 16485790 /tmp/test-18658-2023-06-07T05-56-09.2573311792264305458/33-client.log
testsvc  18658  test   92u      REG               0,40       977 16520841 /tmp/test-18658-2023-06-07T05-56-09.2573311792264305458/194-client.log
testsvc  18658  test   93u      REG               0,40       977 16509370 /tmp/test-18658-2023-06-07T05-56-09.2573311792264305458/177-client.log
testsvc  18658  test   94u      REG               0,40       254 16522704 /tmp/test-18658-2023-06-07T05-56-09.2573311792264305458/195-client.log
testsvc  18658  test   95u      REG               0,40       518 16515177 /tmp/test-18658-2023-06-07T05-56-09.2573311792264305458/178-client.log
testsvc  18658  test   96u      REG               0,40       518 16485297 /tmp/test-18658-2023-06-07T05-56-09.2573311792264305458/96-client.log
testsvc  18658  test   97u      REG               0,40       977 16534629 /tmp/test-18658-2023-06-07T05-56-09.2573311792264305458/196-client.log
testsvc  18658  test   98u      REG               0,40        81 16535696 /tmp/test-18658-2023-06-07T05-56-09.2573311792264305458/199-client.log
testsvc  18658  test   99u      REG               0,40       518 16497763 /tmp/test-18658-2023-06-07T05-56-09.2573311792264305458/73-client.log
testsvc  18658  test  100u      REG               0,40       436 19855382 /tmp/test-18658-2023-06-07T05-56-09.2573311792264305458/4-11192.req.xml
testsvc  18658  test  101u      REG               0,40        78 16490665 /tmp/test-18658-2023-06-07T05-56-09.2573311792264305458/47-client.log
testsvc  18658  test  102u      REG               0,40        88 16476929 /tmp/test-18658-2023-06-07T05-56-09.2573311792264305458/39-client.log
testsvc  18658  test  103u      REG               0,40       977 16486100 /tmp/test-18658-2023-06-07T05-56-09.2573311792264305458/72-client.log
testsvc  18658  test  104u      REG               0,40        81 16526211 /tmp/test-18658-2023-06-07T05-56-09.2573311792264305458/198-client.log
testsvc  18658  test  105u      REG               0,40       977 16522857 /tmp/test-18658-2023-06-07T05-56-09.2573311792264305458/200-client.log
testsvc  18658  test  106u      REG               0,40        88 16486766 /tmp/test-18658-2023-06-07T05-56-09.2573311792264305458/40-client.log
testsvc  18658  test  107u      REG               0,40        78 16483311 /tmp/test-18658-2023-06-07T05-56-09.2573311792264305458/49-client.log
testsvc  18658  test  108u      REG               0,40       977 16538794 /tmp/test-18658-2023-06-07T05-56-09.2573311792264305458/201-client.log
testsvc  18658  test  109u      REG               0,40       152 16477098 /tmp/test-18658-2023-06-07T05-56-09.2573311792264305458/51-client.log
testsvc  18658  test  110u      REG               0,40        88 16462812 /tmp/test-18658-2023-06-07T05-56-09.2573311792264305458/41-client.log
testsvc  18658  test  111u      REG               0,40       518 16508834 /tmp/test-18658-2023-06-07T05-56-09.2573311792264305458/190-client.log
testsvc  18658  test  112u      REG               0,40        78 16490672 /tmp/test-18658-2023-06-07T05-56-09.2573311792264305458/48-client.log
testsvc  18658  test  113u      REG               0,40       977 17612582 /tmp/test-18658-2023-06-07T05-56-09.2573311792264305458/1359-client.log
testsvc  18658  test  114u      REG               0,40         0 16477111 /tmp/test-18658-2023-06-07T05-56-09.2573311792264305458/54-client.log
```

**Additional context**
The creation of a file per client/request/response seems unnecessary wasteful. It feels like this could be consolidated somehow into a single file that can just be grepped.

**Comment by dougm** [comment]
Should we plug the leak or just remove the client.log file? I think it was used in the early days to make some govc optimizations (such as caching of ServiceContent and sessions). Not sure if those method call durations are still considered useful?

Note that it is possible to send all trace to a single file. For example, this "provider" impl sends everything to stderr:
https://github.com/vmware/govmomi/blob/main/vim25/debug/log.go#L24

**Comment by brakthehack** [comment]
I don't have any experience using this feature. But, maybe it makes sense to use the same patterns across providers. E.g. the log provider writes to a single file so maybe the file provider should too? It seems a bit misleading that the `NewFile` method in the log provider swallows the parameter.

**Comment by dougm** [comment]
> It seems a bit misleading that the NewFile method in the log provider swallows the parameter
True, the original interface didn't consider the single-file / stderr use-case, it was focused on tracing many govc instances running in parallel.  We didn't have this notion of "service client" back then either, only had the vim25 endpoint.  We can revisit the debug/file interface, but in the short-term we can just remove the client.log. I'll open a PR for that today. Thanks for tracking this leak down @brakthehack
